core/src/test/java/org/testcontainers/junit/DependenciesTest.java

```diff
@@ -0,0 +1,116 @@
+        VisibleAssertions.assertEquals("Does not trigger .stop()", 0, startable.getStopInvocationCount().intValue());
+    }
+
+    @Test
+    public void shouldWorkWithMutlipleDependencies() {
+        InvocationCountingStartable startable1 = new InvocationCountingStartable();
+        InvocationCountingStartable startable2 = new InvocationCountingStartable();
+
+        try (
+            GenericContainer container = new GenericContainer()
+                .withStartupCheckStrategy(new OneShotStartupCheckStrategy())
+                .dependsOn(startable1, startable2)
+        ) {
+            container.start();
+        }
+
+        VisibleAssertions.assertEquals("Startable1 started once", 1, startable1.getStartInvocationCount().intValue());
+        VisibleAssertions.assertEquals("Startable2 started once", 1, startable2.getStartInvocationCount().intValue());
+    }
+
+    @Test
+    public void shouldStartEveryTime() {
+        InvocationCountingStartable startable = new InvocationCountingStartable();
+
+        try (
+            GenericContainer container = new GenericContainer()
+                .withStartupCheckStrategy(new OneShotStartupCheckStrategy())
+                .dependsOn(startable)
+        ) {
+            container.start();
+            container.stop();
+
+            container.start();
+            container.stop();
+
+            container.start();
+        }
+
+        VisibleAssertions.assertEquals("Started multiple times", 3, startable.getStartInvocationCount().intValue());
+        VisibleAssertions.assertEquals("Does not trigger .stop()", 0, startable.getStopInvocationCount().intValue());
+    }
+
+    @Test
+    public void shouldStartTransitiveDependencies() {
+        InvocationCountingStartable transitiveOfTransitiveStartable = new InvocationCountingStartable();
+        InvocationCountingStartable transitiveStartable = new InvocationCountingStartable();
+        transitiveStartable.getDependencies().add(transitiveOfTransitiveStartable);
+
+        InvocationCountingStartable startable = new InvocationCountingStartable();
+        startable.getDependencies().add(transitiveStartable);
+
+        try (
+            GenericContainer container = new GenericContainer()
+                .withStartupCheckStrategy(new OneShotStartupCheckStrategy())
+                .dependsOn(startable)
+        ) {
+            container.start();
+            container.stop();
+        }
+
+        VisibleAssertions.assertEquals("Root started", 1, startable.getStartInvocationCount().intValue());
+        VisibleAssertions.assertEquals("Transitive started", 1, transitiveStartable.getStartInvocationCount().intValue());
+        VisibleAssertions.assertEquals("Transitive of transitive started", 1, transitiveOfTransitiveStartable.getStartInvocationCount().intValue());
+    }
+
+    private static class InvocationCountingStartable implements Startable {
+
+        @Getter
```

Comment:
Absolutely no big deal, but this is a private test class - do we really need the formality of getters? 😁

Reply:
just a matter of habit and to keep it consistent with the rest of the code :D